### PR TITLE
feature6

### PR DESCRIPTION
**Pull request by EmomaxD**

### CODE REVIEW

File: editor-components/feature/feature6/feature6.module.scss [hunk 1]
```diff
@@ -1,60 +1,126 @@
...
-                    min-height: 200px;
-                    padding: 20px;
+                    height: max-content;
+                    padding: 0;
                     flex: 1 1;
-                    align-items: center;
-                    .title {
+                    margin: 24px;
+                    margin-top: 55px;
+                    overflow: hidden; /* Added to clip the scaled image */
```
Comment (Ayazelar) [inline editor-components/feature/feature6/feature6.module.scss:45]:
```suggestion
                    overflow: hidden;
```

File: editor-components/feature/feature6/feature6.module.scss [hunk 2]
```diff
@@ -1,60 +1,126 @@
...
-                            white-space: nowrap;
-                            vertical-align: middle;
-                            font-size: 1.2rem;
-                            font-weight: 100;
-                            color: #fff;
-                            background: var(--composer-secondary-color);
-                            text-align: center;
-                            border-radius: 3px;
-                        }
+                        transition: transform .3s ease; /* Added ease transition */
```
Comment (Ayazelar) [inline editor-components/feature/feature6/feature6.module.scss:50]:
```suggestion
                        transition: transform .3s ease; 
```

File: editor-components/feature/feature6/feature6.module.scss [hunk 3]
```diff
@@ -1,60 +1,126 @@
...
+                        object-fit: cover;
+                        text-align: center;
                     }
-                    .long-text {
+                    .image:hover {
+                        transform: scale(1.1);
+                    }
+                    .image-shadow {
+                        position: absolute;
+                        top: 55%; /* Adjusted to cover the bottom 45% */
```
Comment (Ayazelar) [inline editor-components/feature/feature6/feature6.module.scss:59]:
```suggestion
                        top: 55%; 
```

File: editor-components/feature/feature6/feature6.module.scss [hunk 4]
```diff
@@ -1,60 +1,126 @@
...
+                        left: 0;
+                        right: 0;
+                        bottom: 0;
+                        pointer-events: none;
+                        display: flex;
+                        align-items: center;
+                        justify-content: center;
+                        width: 100%;
+                        height: 45%; /* Adjusted to cover the bottom 45% */
```
Comment (Ayazelar) [inline editor-components/feature/feature6/feature6.module.scss:68]:
```suggestion
                        height: 45%;
```

File: editor-components/feature/feature6/feature6.module.scss [hunk 5]
```diff
@@ -63,9 +129,43 @@
     .container {
         padding: 50px 20px;
         .max-content {
-            .card-item-count {
-                min-width: 100%;
+            .content {
+                gap: 20px !important; /* Decreased gap for smaller screens */
```
Comment (Ayazelar) [inline editor-components/feature/feature6/feature6.module.scss:133]:
```suggestion
                gap: 20px !important;
```

File: editor-components/feature/feature6/feature6.module.scss [hunk 6]
```diff
@@ -63,9 +129,43 @@
...
+                .header-title {
+                    font-size: 28px;
+                }
+                .header-subtitle {
+                    font-size: 16px;
+                }
+                .button-position {
+                    flex-direction: column;
+                    overflow: hidden;
+                    gap: 5px; /* Decreased gap for buttons on smaller screens */
```
Comment (Ayazelar) [inline editor-components/feature/feature6/feature6.module.scss:150]:
```suggestion
                    gap: 5px;
```

File: editor-components/feature/feature6/feature6.tsx [hunk 7]
```diff
@@ -1,116 +1,105 @@
...
             {
-              type: "string",
-              key: "title",
-              displayer: "Title",
+              type: "image",
+              key: "image",
+              displayer: "Image",
               value:
-                "Scientists Discover New Species of Butterfly in the Amazon Rainforest",
+                "https://demo.tagdiv.com/newspaper_interior_designer_pro/wp-content/uploads/2023/11/10-3-696x928.jpg",
```
Comment (Ayazelar) [inline editor-components/feature/feature6/feature6.tsx:56]: should use spica storage url

File: editor-components/feature/feature6/feature6.tsx [hunk 8]
```diff
@@ -1,116 +1,105 @@
...
             {
-              type: "string",
-              key: "title",
-              displayer: "Title",
+              type: "image",
+              key: "image",
+              displayer: "Image",
               value:
-                "The Benefits of Meditation for Mental and Physical Health",
+                "https://demo.tagdiv.com/newspaper_interior_designer_pro/wp-content/uploads/2023/11/9-3-696x614.jpg",
```
Comment (Ayazelar) [inline editor-components/feature/feature6/feature6.tsx:76]: should use spica storage url

File: editor-components/feature/feature6/feature6.tsx [hunk 9]
```diff
@@ -1,116 +1,105 @@
...
             {
-              type: "string",
-              key: "title",
-              displayer: "Title",
+              type: "image",
+              key: "image",
+              displayer: "Image",
               value:
-                "The Importance of Time Management for Increased Productivity",
+                "https://demo.tagdiv.com/newspaper_interior_designer_pro/wp-content/uploads/2023/11/50-3-696x510.jpg"
```
Comment (Ayazelar) [inline editor-components/feature/feature6/feature6.tsx:96]: should use spica storage url

**Review by Ayazelar — CHANGES_REQUESTED** [review]
.

File: editor-components/feature/feature6/feature6.tsx [hunk 10]
```diff
@@ -131,26 +151,43 @@ class Feature6 extends BaseFeature {
   render() {
     return (
       <div className={this.decorateCSS("container")}>
-        <div className={this.decorateCSS("max-content")}>
-          <div className={this.decorateCSS("content")}>
-            {this.castToObject<IBasicBadge[]>("basic-card-badge").map(
-              (basic: any, index: number) => (
+    <div className={this.decorateCSS("max-content")}>
+        <div className={this.decorateCSS("content")}>
+            <div className={this.decorateCSS("header-title")}>{this.getPropValue("header")}</div>
```
Comment (dogusmiuw) [inline editor-components/feature/feature6/feature6.tsx:156]: If no title entered, remove its relevant elements. Do this for all.

File: editor-components/feature/feature6/feature6.tsx [hunk 11]
```diff
@@ -1,116 +1,105 @@
...
-type IBasicBadge = {
+
+type Card = {
+  image: string;
   title: string;
-  description: string;
-  badge: string;
 };
+
+type Button = {
```
Comment (dogusmiuw) [inline editor-components/feature/feature6/feature6.tsx:12]: Unused code. Remove it.

File: editor-components/feature/feature6/feature6.tsx [hunk 12]
```diff
@@ -121,6 +110,37 @@ class Feature6 extends BaseFeature {
...
+    };
+
+    this.addProp({
+      type: "array",
+      key: "buttons",
+      displayer: "Buttons",
+      value: [JSON.parse(JSON.stringify(button))],
+      additionalParams: {
+        maxElementCount:  2
+      }
```
Comment (dogusmiuw) [inline editor-components/feature/feature6/feature6.tsx:143]: Can you explain this section? Why did you choose this way? And why do you need it?

**Review by dogusmiuw — CHANGES_REQUESTED** [review]
Please handle the changes.

File: editor-components/feature/feature6/feature6.tsx [hunk 13]
```diff
@@ -1,116 +1,105 @@
 import * as React from "react";
-import { BaseFeature } from "../../EditorComponent";
+import { BaseFeature, TypeUsableComponentProps } from "../../EditorComponent";
```
Comment (dogusmiuw) [inline editor-components/feature/feature6/feature6.tsx:2]: remove TypeUsableComponentProps from this import

File: editor-components/feature/feature6/feature6.tsx [hunk 14]
```diff
@@ -129,24 +169,91 @@ class Feature6 extends BaseFeature {
...
+                            {card.image && (
+                              <img
+                                className={this.decorateCSS("image")}
+                                src={card.image}
+                                alt="feature"
+                              />
+                            )}
+                            <div
+                              style={!card.image ? { height: "100%" } : {}}
+                              className={`
```
Comment (Ayazelar) [inline editor-components/feature/feature6/feature6.tsx:223]: use className instead inline style

**Review by Ayazelar — CHANGES_REQUESTED** [review]
.

**Review by Ayazelar — APPROVED** [review]

**Review by dogusmiuw — APPROVED** [review]

**Pull request merged**